app/shared/components/dynamic-content/dynamic-content-details.component.ts

```diff
@@ -1,41 +1,77 @@
-  styles: [
-    <string> require('./dynamic-content-details.component.styl')
-  ],
-  pipes: [ToDate]
+  template: require('./dynamic-content-details.component.html') as string,
+  directives: [EntriesViewComponent, SidebarComponent, LineSocialComponent, RouterLink, Angulartics2On, TagsComponent],
+  styles: [require('./dynamic-content-details.component.styl') as string],
+  pipes: [ToDatePipe]
 })
-@CanActivate(checkContentType)
-export class DynamicComponentDetails implements OnActivate {
+export class DynamicContentDetailsComponent implements OnActivate, CanReuse {
   private content: NodePageContent;
-  private contentType: string;
+  private childList: ContentfulNodePage[];
+  private slug: any;
+  private routesGatewayService: RoutesGatewayService;
+  private urlPath: string;
+  private contentSlug: string;
+  private router: Router;
+  private contentfulContent: ContenfulContent;
+  private breadcrumbsService: BreadcrumbsService;
 
-  constructor(private _contentfulContent: ContenfulContent, private _router: Router) {
+  public constructor(@Inject(Router) router: Router,
+                     @Inject(ContenfulContent) contentfulContent: ContenfulContent,
+                     @Inject(BreadcrumbsService) breadcrumbsService: BreadcrumbsService) {
+    this.router = router;
+    this.contentfulContent = contentfulContent;
+    this.breadcrumbsService = breadcrumbsService;
+
+    // FIXME: Workaround for now - should be injected in a proper way not using the injector explicitly
+    this.routesGatewayService = appInjector().get(RoutesGatewayService);
   }
 
-  routerOnActivate(next: ComponentInstruction, prev: ComponentInstruction): void {
-    this.contentType = (next.params as DynamicContentRouteParams).contentType;
-    this._contentfulContent
-      .getNodePage((next.params as DynamicContentDetailRouteParams).contentSlug)
+  public routerOnActivate(next: ComponentInstruction): any {
+    this.urlPath = next.urlPath;
+    this.contentSlug = next.urlPath.split('/').pop();
+    this.contentfulContent
+      .getNodePage(this.contentSlug)
       .subscribe(
-        content => {
+        (content: ContentfulNodePage[]) => {
           if (!content) {
-            this._router.navigate(['Root']);
+            this.router.navigate(['Root']);
           }
-          this.content = content
-        }
-      );
+          this.content = content[0].fields;
+          this.breadcrumbsService.breadcrumbs$.next({url: next.urlPath, name: this.content.title});
+          this.contentfulContent
+            .getChildrenOf(content[0].sys.id)
+            .subscribe(
+              (child: ContentfulNodePage[]) => {
+                this.childList = child;
+                for (let item of child) {
+                  if (item.fields) {
+                    let title = item.fields.title;
+                    this.slug = `${this.urlPath}/${item.fields.slug}`;
+                    this.routesGatewayService.addRoute(this.slug, {name: title});
```

Comment:
this.routesGatewayService.addRoute(`${this.urlPath}/${item.fields.slug}`, {name: title});